### PR TITLE
Feature/ker/ported changes

### PR DESCRIPTION
**Pull request by keriksson-rosenqvist**
Adding hardcoded delay to qubit drives and counting two clock cycles per instruction for the sequencer.
squash phase shifts
switch hive fix
Backport instruction serialization
Fix frequency frame attribute access.
Add support for base64 encoded bitcode through the front end. Tweak measure variable assignment.
OpenPulse Tidy

### CODE REVIEW

**Comment by keriksson-rosenqvist** [comment]
I have read the Contributor License Agreement and I hereby accept the Terms.

File: src/QAT/qat/purr/backends/live.py [hunk 1]
```diff
@@ -248,46 +248,42 @@ def optimize(self, instructions):
 
             return instructions
 
+    def validate_hardware(self, instructions: List[Instruction]):
```
Comment (chemix-lunacy) [inline src/QAT/qat/purr/backends/live.py:251]: This shouldn't really be a new method unless it's used somewhere I can't see. Whenever you're validating you want to run everything anyway, including the parent method, so I'd like to see it moved back unless there's a strong reason for having a new public one.

File: src/QAT/qat/purr/compiler/execution.py [hunk 2]
```diff
@@ -419,8 +453,13 @@ def process_pulse(
             start=-centre + 0.5 * dt, stop=length - centre - 0.5 * dt, num=samples
         )
         pulse = evaluate_shape(position.instruction, t, phase)
-        if not position.instruction.ignore_channel_scale:
-            pulse *= pulse_channel.scale
+
+        scale = pulse_channel.scale
+        if isinstance(
+            position.instruction, (Pulse, CustomPulse)
+        ) and position.instruction.ignore_channel_scale:
```
Comment (chemix-lunacy) [inline src/QAT/qat/purr/compiler/execution.py:460]: Not too important but formatting a little wonky.

File: src/QAT/qat/purr/backends/live.py (hunk 1 above)
Comment (keriksson-rosenqvist) [reply src/QAT/qat/purr/backends/live.py]: Reverted

**Review by chemix-lunacy — APPROVED** [review]

**Review by hamidelmaazouz — APPROVED** [review]

**Pull request merged**